### ISSUE TITLE
Could this make standalone exe which dont need any external files?

### ISSUE THREAD

**Issue by ruthan** [SuperIlu/DOjS#9]
Hello,
i wonder if Could this make standalone exe which dont need any external files?

 If yes, which versions of MS-DOS are supported and what is size and memory footprint of minimal hello world?

**Comment by SuperIlu** [comment]
Just to clarify:
1. You want DOjS to be a standalone EXE? or
1. You want DOjS to "create" a standalone EXE with the script included?

So far both is impossible. DOjS relies on the files in `jsboot/` (for constants/functions which are implemented in Javascript) and on `GLIDE3X.DXE` and `CWSDPMI.EXE` (for 3dfx support and DPMI).

**Comment by ruthan** [comment]
Second one. 

Dos has possiblity to pack multiple files into *.exe as .NET has resources so you can pack files into, or extract them after its start to some temp.
  But GLIDE3X and CWSDPMI are fine, even as external.. important is that user can run all by one exe and dont need adjust its installation (install some runtime, add something to path etc.).. but only run exe.

**Comment by SuperIlu** [comment]
For now this has no priority for me as you can provide the end user with a simple .BAT to run any script without deeper knowledge of DOjS.
If anybody knows more about creating single file EXE and can provide contributions I will happily include it...

**Comment by robilic** [comment]
In the old days of DJGPP you could use CWSDSTUB.EXE

_CWSDSTUB.EXE is a stub loader image for DJGPP which includes CWSDPMI.  This allows single executable image distributions.  You can use the EXE2COFF program and COPY /B CWSDSTUB.EXE+yourimage yourimage.exe to create a standalone executable image._

http://ctan.imsc.res.in/obsolete/systems/msdos/pdftex/cwsdpmi/BIN/CWSDPMI.DOC

**Comment by SuperIlu** [comment]
This still leaves the problem that you need the appropriate 3dfx `GLIDE3X.DXE`, everything below `JSBOOT/` and the script you want to run besides `DOJS.EXE`.

**Comment by SuperIlu** [comment]
Update on this issue:
I'm working on ZIP file support right now. In the future it will be possible to include (nearly) all needed files in a ZIP file.
This will not result in a standalone EXE, but the number of files will be reduced to ~3 files in total:

- DOjS.EXE
- GLIDE3X.DXE
- ZIP file with everything else

**Comment by ruthan** [comment]
What about code itselft it be 4 file, or it would be integrated it some other? Sorry, im still waiting to this fix, to start use DojS, so i dont know details. 
  Execution would dojS.exe - someJavaScript.js  or its possible make some start.exe for Exe solution? I dunno how in dos, but for Windows exist some Bat to Exe convertor.

**Comment by cyberluke** [comment]
Why closed after one year with no comment? This was a crucial point for adoption by many people at Vogons…

**Comment by ruthan** [comment]
Yeah this is really important, and GLide shouldnt be mandatory.. Some standalone.exe with some data archive with everything else is still better than nothing. BTW some clever detection of 3dfx card type and Glide dxe autoselection would be nice too.
